pkg/factory/factory.go

```diff
@@ -177,37 +177,64 @@ func (rf *resourceFactory) GetDeviceProvider(dt types.DeviceType) types.DevicePr
 	}
 }
 
-// GetDeviceFilter unmarshal the "selector" values from ResourceConfig and returns an instance of DeviceSelector based on
+// GetDeviceFilter unmarshal the "selector" values from ResourceConfig and returns a slice of *DeviceSelectors based on
 // DeviceType in the ResourceConfig
-func (rf *resourceFactory) GetDeviceFilter(rc *types.ResourceConfig) (interface{}, error) {
+func (rf *resourceFactory) GetDeviceFilter(rc *types.ResourceConfig) ([]interface{}, error) {
 	switch rc.DeviceType {
 	case types.NetDeviceType:
-		netDeviceSelector := &types.NetDeviceSelectors{}
+		netDeviceArray := make([]types.NetDeviceSelectors, 1)
```

Comment:
nit: netDeviceSelectors ? here and below

technically you are creating a slice (there are no arrays in go).
also we generally dont add the type in the name (e.g[ hungarian style](https://en.wikipedia.org/wiki/Hungarian_notation))